packages/@react-spectrum/table/test/Table.test.js

```diff
@@ -4206,4 +4137,158 @@ describe('TableView', function () {
+      let onSelectionChange = jest.fn();
+      let tree = render(
+        <div>
+          <TableView aria-label="Table" selectionMode="multiple" onSelectionChange={onSelectionChange} renderEmptyState={() => <h3>No results</h3>}>
+            <TableHeader>
+              <Column key="foo">Foo</Column>
+              <Column key="bar">Bar</Column>
+            </TableHeader>
+            <TableBody>
+              {[]}
+            </TableBody>
+          </TableView>
+          <input />
+        </div>
+      );
+
+      let table = tree.getByRole('grid');
+      let selectAll = tree.getByRole('checkbox');
+
+      userEvent.tab();
+      expect(document.activeElement).toBe(table);
+      userEvent.tab();
+      expect(document.activeElement).not.toBe(selectAll);
+      expect(selectAll).toHaveAttribute('disabled');
+    });
+
+    it('should allow the user to tab into the table body', function () {
+      let tree = render(<EmptyStateTable />);
+      let table = tree.getByRole('grid');
+      let toggleButton = tree.getAllByRole('button')[0];
+      let link = tree.getByRole('link');
+
+      userEvent.tab();
+      expect(document.activeElement).toBe(toggleButton);
+      userEvent.tab();
+      expect(document.activeElement).toBe(table);
+      userEvent.tab();
+      expect(document.activeElement).toBe(link);
+    });
+
+    it('should disable keyboard navigation within the table', function () {
+      let tree = render(<EmptyStateTable />);
+      let table = tree.getByRole('grid');
+      let header = within(table).getAllByRole('columnheader')[2];
+      let headerButton = within(header).getByRole('button');
+      expect(headerButton).toHaveAttribute('aria-disabled', 'true');
+      // Programatically focus the column header since we can't tab to it
+      act(() => {
+        header.focus();
+      });
+      expect(document.activeElement).toBe(header);
+
+      fireEvent.keyDown(document.activeElement, {key: 'ArrowLeft', code: 37, charCode: 37});
+      fireEvent.keyUp(document.activeElement, {key: 'ArrowLeft', code: 37, charCode: 37});
+      expect(document.activeElement).toBe(header);
+    });
+
+    it('should disable press interactions with the column headers', function () {
+      let tree = render(<EmptyStateTable />);
+      let table = tree.getByRole('grid');
+      let headers = within(table).getAllByRole('columnheader');
+      let toggleButton = tree.getAllByRole('button')[0];
+
+      userEvent.tab();
+      expect(document.activeElement).toBe(toggleButton);
+
+      let columnButton = within(headers[2]).getByRole('button');
+      triggerPress(columnButton);
+      expect(document.activeElement).toBe(toggleButton);
+      expect(tree.queryByRole('menuitem')).toBeFalsy();
+      fireEvent.mouseEnter(headers[2]);
+      act(() => {jest.runAllTimers();});
+      expect(tree.queryByRole('slider')).toBeFalsy();
+    });
+
+    it.skip('should re-enable functionality when the table recieves items', function () {
```

Comment:
As mentioned, skipped for now until https://github.com/adobe/react-spectrum/pull/3414 goes in